clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/common/cache/Keys.groovy

```diff
@@ -46,7 +46,12 @@ class Keys {
     }
   }
 
+
   static Map<String, String> parse(AzureCloudProvider azureCloudProvider, String key) {
+    parse(azureCloudProvider.id, key)
+  }
+
+  static Map<String, String> parse(String providerId, String key){
```

Comment:
you can probably ditch the providerId here and only handle items where the id segment of the key == `AzureCloudProvider.AZURE`